src/main/java/io/axoniq/axonserver/connector/event/DcbEventChannel.java

```diff
@@ -0,0 +1,151 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.axoniq.axonserver.connector.event;
+
+import io.axoniq.axonserver.connector.ResultStream;
+import io.axoniq.axonserver.grpc.event.dcb.AppendEventsResponse;
+import io.axoniq.axonserver.grpc.event.dcb.ConsistencyCondition;
+import io.axoniq.axonserver.grpc.event.dcb.GetHeadRequest;
+import io.axoniq.axonserver.grpc.event.dcb.GetHeadResponse;
+import io.axoniq.axonserver.grpc.event.dcb.GetSequenceAtRequest;
+import io.axoniq.axonserver.grpc.event.dcb.GetSequenceAtResponse;
+import io.axoniq.axonserver.grpc.event.dcb.GetTagsRequest;
+import io.axoniq.axonserver.grpc.event.dcb.GetTagsResponse;
+import io.axoniq.axonserver.grpc.event.dcb.GetTailRequest;
+import io.axoniq.axonserver.grpc.event.dcb.GetTailResponse;
+import io.axoniq.axonserver.grpc.event.dcb.SourceEventsRequest;
+import io.axoniq.axonserver.grpc.event.dcb.SourceEventsResponse;
+import io.axoniq.axonserver.grpc.event.dcb.StreamEventsRequest;
+import io.axoniq.axonserver.grpc.event.dcb.StreamEventsResponse;
+import io.axoniq.axonserver.grpc.event.dcb.TaggedEvent;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Communication channel for Event related interaction with Axon Server based on Dynamic Consistency Boundaries
+ * concept.
+ *
+ * @author Milan Savic
+ * @since 2024.1.0
+ */
+public interface DcbEventChannel {
+
+    /**
+     * Starts a new transaction to append events.
+     *
+     * @return the transaction reference onto which to register events to append along with the condition
+     */
+    AppendEventsTransaction startTransaction();
+
+    /**
+     * Provides an infinite stream of events.
+     *
+     * @param request the request used to filter events
+     * @return an infinite stream of events
+     */
+    ResultStream<StreamEventsResponse> stream(StreamEventsRequest request);
+
+    /**
+     * Provides a finite stream of events used to eventsource a model.
+     *
+     * @param request the query used to filter out events for sourcing
+     * @return the response containing events to source a model and a consistency marker to be used when trying to
+     * append new events to the event store
+     */
+    ResultStream<SourceEventsResponse> source(SourceEventsRequest request);
+
+    /**
+     * Provides tags for an event at the given global sequence.
+     *
+     * @param request the sequence of an event
+     * @return the tags fo the event
+     */
+    CompletableFuture<GetTagsResponse> tagsFor(GetTagsRequest request);
+
+    /**
+     * Provides the head of the event store.
+     *
+     * @param request empty request
+     * @return the head of the event store
+     */
+    CompletableFuture<GetHeadResponse> head(GetHeadRequest request);
+
+    /**
+     * Provides the tail of the event store.
+     *
+     * @param request empty request
```

Comment:
Just "empty request" isn't super clear. What if we do something like this?

```suggestion
     * @param request empty request, although required for the operation regardless.
```

Reply:
Now that I think about it, given that the request object is empty anyhow, why not omit it entirely? The implementation can simply make that request at all times.